tests/e2e/k8s_test.go

```diff
@@ -0,0 +1,174 @@
+func currentEnvK8sContext() (kubeconfigPath string, kubecontextName string, err error) {
+	kubeconfigPath, isExisting := os.LookupEnv("KUBECONFIG")
+	if !isExisting {
+		homePath, err := os.UserHomeDir()
+		if err != nil {
+			return "", "", errors.WrapIf(err, "retrieving user home directory failed")
+		}
+
+		kubeconfigPath = path.Join(homePath, ".kube", "config")
+	}
+
+	kubeconfigBytes, err := os.ReadFile(kubeconfigPath)
+	if err != nil {
+		return "", "", errors.WrapIfWithDetails(err, "reading KUBECONFIG file failed", "path", kubeconfigPath)
+	}
+
+	structuredKubeconfig := make(map[string]interface{})
+	err = yaml.Unmarshal(kubeconfigBytes, &structuredKubeconfig)
+	if err != nil {
+		return "", "", errors.WrapIfWithDetails(
+			err,
+			"parsing kubeconfig failed",
+			"kubeconfig", string(kubeconfigBytes),
+		)
+	}
+
+	kubecontext, isOk := structuredKubeconfig["current-context"].(string)
+	if !isOk {
+		return "", "", errors.WrapIfWithDetails(
+			err,
+			"kubeconfig current-context is not string",
+			"current-context", structuredKubeconfig["current-context"],
+		)
+	}
+
+	return kubeconfigPath, kubecontext, nil
+}
+
+// listK8sCRDs lists the available CRDs from the specified kubectl context and
+// namespace optionally filtering for the specified CRD names.
+func listK8sCRDs(kubectlOptions *k8s.KubectlOptions, crdNames ...string) []string {
+	if len(crdNames) == 0 {
+		By("Listing CRDs")
+	} else {
+		By(fmt.Sprintf("Listing CRDs filtered for CRD names %+v", crdNames))
+	}
+
+	args := append([]string{"get", "crd", "-o", "name"}, crdNames...)
+	output, err := k8s.RunKubectlAndGetOutputE(
+		GinkgoT(),
+		kubectlOptions,
+		args...,
+	)
+
+	Expect(err).NotTo(HaveOccurred())
+
+	return strings.Split(output, "\n")
+}
+
+// replaceK8sResourcesFromManifest replaces existing Kubernetes resources from
+// the specified manifest to the provided kubectl context and namespace.
+func replaceK8sResourcesFromManifest(kubectlOptions *k8s.KubectlOptions, manifestPath string, shouldBeValidated bool) {
+	By(fmt.Sprintf("Replacing k8s resources from manifest %s", manifestPath))
+	k8s.RunKubectl(
+		GinkgoT(),
+		kubectlOptions,
+		"replace",
+		fmt.Sprintf("--validate=%t", shouldBeValidated),
+		"--filename", manifestPath,
+	)
+}
+
+// requireExistingCRDs checks whether the specified CRDs are existing on
+// the provided kubectl context.
+func requireExistingCRDs(kubectlOptions *k8s.KubectlOptions, crdNames ...string) {
+	crds := listK8sCRDs(kubectlOptions, crdNames...)
+
+	crdFullNames := make([]string, 0, len(crds))
+	for _, crdName := range crdNames {
+		crdFullNames = append(crdFullNames, "customresourcedefinition.apiextensions.k8s.io/"+crdName)
+	}
+
+	Expect(crds).To(ContainElements(crdFullNames))
+}
+
+// requireRunningPods checks whether the specified pod names are existing in the
+// namespace and have a running status.
+func requireRunningPods(kubectlOptions *k8s.KubectlOptions, matchingLabelKey string, podNames ...string) {
```

Comment:
Also this function doesn't really work in some cases, for example if I'm trying to check a kafka cluster's pods that are in the kafka namespace this will fail because the number of pods in the kafka namespace don't match the length of podNames (because of the kafka-operator pod being in that namespace). Also our kafka cluster pods don't really have any label with the pod name itself so that's another thing that won't really work as expected.
Another thing: Since we're checking a label I don't really feel like we're doing what we say we're doing in the function name and description either.

Reply:
While I agree we can use higher level functions to check more of a healthy state, I would definitely separate checking 
- resources we recently interacted with and 
- resources which should remain idle/functioning
to make sure in case of a failure we have a quick understanding of whether the resources failed the healthiness check which we manipulated or something unrelated broke so far unexpectedly.

The description definitely requires an update.
I take suggestions for a better function name, I didn't want to involve health/ready keywords unless we check the readiness/liveness endpoints of the pods.

Reply:
Im thinking on to only check the readiness of the pod like here: https://github.com/banzaicloud/koperator/blob/f7fe06f20be0d303c28575169aaaa0ef1f0d1888/tests/e2e/kafka_prod_consum_test.go#L51
When it fails we put the kubectl describe pod output into the error

Reply:
I decided to remove this function, because it was only used in the Helm post install pod readiness check and that was removed in https://github.com/banzaicloud/koperator/commit/eb366989392cb596456685145f47b9612bc54a2a. see https://github.com/banzaicloud/koperator/pull/977#discussion_r1208007607 for more details.